src/datatypes/numeric.js

```diff
@@ -66,6 +82,7 @@ function generateFunctions (bufferReader, bufferWriter, size, schema) {
     }
   }
   const writer = (value, buffer, offset) => {
+    if (typeof value === 'bigint') value = Number(value) // eslint-disable-line
```

Comment:
Bug ahead - accuracy after `Number.MAX_SAFE_INTEGER` will be lost

Reply:
64-bit integers wouldn't be written here, so there's no issue of overflow

Reply:
Why would you leave functions, when you can add the following lines to `nums`:
```
  'i64': ['readBigInt64BE', 'writeBigInt64BE', 8],
  'u64': ['readBigUInt64BE', 'writeBigUInt64BE', 8],
  'li64': ['readBigInt64LE', 'writeBigInt64LE', 8],
  'lu64': ['readBigUInt64LE', 'writeBigUInt64LE', 8],
```

Reply:
That would be a breaking change

Reply:
Isn't that already a breaking change?

Reply:
@extremeheat Okay, I found a way how to keep two behaviors at same time
```js
class BigIntExtended {
    constructor (arg, isLE) {
        this.bint = BigInt(arg)
        if (isLE) {
            this[0] = this.upper
            this[1] = this.lower
        } else {
            this[0] = this.lower
            this[1] = this.upper
        }
    }
    get lower () { return Number(this.bint & 0xFFFFFFFFn) }
    get upper () { return Number((this.bint >> 32n) & 0xFFFFFFFFn) }
    valueOf () { return this.bint }
}
```

Reply:
Yeah, this seems interesting. So in order to write, the check would be typeof var.valueOf() instead, to read it would read  as a BigInt and return this class. Also, a toJSON function could be added to make it serializable. Would be particularly useful for https://github.com/PrismarineJS/mineflayer/pull/1543 I think. cc opinion of @rom1504 @Karang

Reply:
@Saiv46 you updated it, why do you do the shifting in valueOf? arg can just be stored and returned by valueOf(), and array access would still work. What do you think about just

```js
class BigIntExtended extends Array {
    constructor (arg, isLE) {
        const arr = [Number((arg >> 32n) & 0xFFFFFFFFn), Number(arg & 0xFFFFFFFFn)]
        if (isLE) arr.reverse()
        super(...arr)
        this.arg = arg
        this.isLE = isLE | 0
        this.valueOf = () => arg
    }
}
```
?